homeassistant/components/hyperion/light.py

```diff
@@ -138,7 +138,7 @@ async def async_setup_platform(
     if not hyperion_id:
         raise PlatformNotReady
 
-    future_unique_id = get_hyperion_unique_id(hyperion_id, instance)
+    future_unique_id = get_hyperion_unique_id(hyperion_id, instance, LIGHT_DOMAIN)
```

Comment:
We don't need to add the platform name to the unique id. The entity registry is aware of both the integration domain and the platform domain.

Reply:
In this case we're using the platform domain as I don't envisage there'll be multiple kinds of light entity. However the idea is that this is a generic name field, it could be "light_secondary", "instance_switch", "clear_priorities_switch", etc. Now felt like the time to worry about this before there are entitys with these unique_ids in the wild.

The alternative is to omit this name field for the main light, but then add the name optionally for future entities as a suffix.

With this change, the unique_ids will be:

```
f9aab089-f85a-55cf-b7c1-222a72faebe9_1_light
f9aab089-f85a-55cf-b7c1-222a72faebe9_1_switch_instance
f9aab089-f85a-55cf-b7c1-222a72faebe9_1_switch_foo
```

Without the change they would be:

```
f9aab089-f85a-55cf-b7c1-222a72faebe9_1
f9aab089-f85a-55cf-b7c1-222a72faebe9_1_switch_instance
f9aab089-f85a-55cf-b7c1-222a72faebe9_1_switch_foo
```

This change proposes that given the choice, the first is better.

Reply:
Don't add the light domain. Add the name of the light class or add a type name to the existing light class that we can use.

Reply:
@MartinHjelmare Not exactly sure I correctly interpreted what you mean -- take a look at the new commit?